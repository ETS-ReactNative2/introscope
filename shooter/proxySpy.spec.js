const { proxySpy, getSpyName } = require('./proxySpy');

const newMock = (v, log = []) => [
    log,
    proxySpy(
        (...args) => log.push(...args),
        'mockName',
        v || function() {} // function has all the Proxy methods available
    )
];

describe('proxySpy', () => {
    it('apply/call', () => {
        const [log, mock] = newMock();
        mock(1, 2, 3);
        expect(log).toMatchSnapshot();
    });

    it('apply/method', () => {
        const [log, mock] = newMock();
        const obj = { method1: mock };
        obj.method1(1, 2, 3);
        expect(log).toMatchSnapshot();
    });
});
