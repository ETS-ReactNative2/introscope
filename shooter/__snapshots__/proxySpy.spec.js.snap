// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proxySpy apply/call 1`] = `
Array [
  Array [
    "call",
    "mockName",
    Array [
      1,
      2,
      3,
    ],
  ],
]
`;

exports[`proxySpy apply/method 1`] = `
Array [
  Array [
    "method",
    "mockName",
    Object {
      "method1": "[Spy mockName]",
    },
    Array [
      1,
      2,
      3,
    ],
  ],
]
`;
